src/GraphQL/DI/GraphQLBuilderBase.cs

```diff
@@ -71,8 +74,24 @@ protected virtual void RegisterDefaultServices()
 
             // configure mapping for IOptions<ErrorInfoProviderOptions>
             Services.Configure<ErrorInfoProviderOptions>();
+
+            // for AOT-compiled scenarios, preserve a few types necessary for DI injection to work properly
+            Preserve<IConfigureExecution[]>();
+            Preserve<IConfigureSchema[]>();
+            Preserve<IGraphTypeMappingProvider[]>();
```

Comment:
I can remove these by adding metadata to the `DocumentExecuter` constructors, probably.